apps/aetestframework/lib/worker.ex

```diff
@@ -1,599 +1,161 @@
-    GenServer.call(__MODULE__, {:sync_two_nodes, node_name1, node_name2})
-  end
-
-  def update_pubkeys_state do
-    GenServer.call(__MODULE__, {:update_pubkeys_state})
-  end
-
-  def get_top_block(node) do
-    GenServer.call(__MODULE__, {:get_top_block, node})
-  end
-
-  def get_top_block_hash(node) do
-    GenServer.call(__MODULE__, {:get_top_block_hash, node})
-  end
+  defstruct port_id: nil, node_port: nil, sync_port: nil
+  use ExConstructor
 
-  def get_oracle_interaction_objects(node) do
-    GenServer.call(__MODULE__, {:get_oracle_interaction_objects, node})
-  end
+  @default_timeout 5_000
 
-  @doc """
-    Updates top block state for 2 nodes and compares them.
-    If top block of each node is equal - they are synced
-  """
-  @spec compare_nodes_by_top_block(String.t(), String.t()) :: :synced | :not_synced
-  def compare_nodes_by_top_block(node_name1, node_name2) do
-    update_node_top_block(node_name1)
-    update_node_top_block(node_name2)
-    block1 = get_top_block(node_name1)
-    block2 = get_top_block(node_name2)
+  # Client API
 
-    if block1 == block2 do
-      :synced
-    else
-      :not_synced
-    end
-  end
-
-  @doc """
-    Updates top block hash state for 2 nodes and compares them.
-    If top block hash of each node is equal - they are synced
-  """
-  @spec compare_nodes_by_top_block_hash(String.t(), String.t()) :: :synced | :not_synced
-  def compare_nodes_by_top_block_hash(node_name1, node_name2) do
-    update_node_top_block_hash(node_name1)
-    update_node_top_block_hash(node_name2)
-    top_block_hash1 = get_top_block_hash(node_name1)
-    top_block_hash2 = get_top_block_hash(node_name2)
-
-    if top_block_hash1 == top_block_hash2 do
-      :synced
-    else
-      :not_synced
-    end
+  def start_link(_args) do
+    GenServer.start_link(__MODULE__, %{}, name: __MODULE__)
   end
 
-  def compare_nodes_by_oracle_interaction_objects(node_name1, node_name2) do
-    update_oracle_interaction_objects(node_name1)
-    update_oracle_interaction_objects(node_name2)
-    oracle_int_obj1 = get_oracle_interaction_objects(node_name1)
-    oracle_int_obj2 = get_oracle_interaction_objects(node_name2)
-
-    if oracle_int_obj1 == oracle_int_obj2 do
-      :synced
-    else
-      :not_synced
-    end
-  end
+  def state, do: GenServer.call(__MODULE__, :state)
 
   @doc """
-    Gets the process ports info.
+  Creates an elixir node
   """
-  def ports_info do
-    GenServer.call(__MODULE__, {:ports_info})
-  end
-
-  @doc """
-    Sending command in a given node.
-  """
-  @spec send_command(String.t(), String.t()) :: :ok | :unknown_node
-  def send_command(node_name, cmd) do
-    # adding \n to cmd to imitate pressing enter in iex shell
-    cmd = cmd <> "\n"
-    GenServer.call(__MODULE__, {:send_command, node_name, cmd}, 10_000)
+  @spec new_node(atom(), non_neg_integer()) :: :ok
+  def new_node(node_name, iex_num) do
+    GenServer.call(__MODULE__, {:new_node, node_name, iex_num}, 10_000)
```

Comment:
I see the timeout is passed as an argument in other places, can you do the same here or just have it as a constant?